assets/src/data/persistence/extrinsic.ts

```diff
@@ -52,10 +52,35 @@ export const readGlobalUserState = async (
     if (lastSet()) {
       await lastSet();
     }
-    const serverUserState = await readGlobal(keys);
-    // merge server state with result
-    if ((serverUserState as any).type !== 'ServerError') {
-      result = serverUserState;
+    let cacheMissingRequestedKey = false;
+    if (keys) {
+      const lastchacheTimeStamp = userStateCache['timestamp'];
```

Comment:
typo, plus you could do like:
`const { timestamp: lastCacheTimeStamp } = userStateCache;`